### PR TITLE
Add system test for `tedge connect c8y --test`

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

Add system tests for `tedge connect c8y --test`.
Two cases, one tests successful case and the other tests failed case.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>
#220

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by rina23q** [comment]
It includes too many unrelated commits. Let me fix...

**Comment by rina23q** [comment]
Removed unrelated commits...

File: tests/PySys/call_tedge_connect_c8y_--test_positive/run.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+When we execute `sudo tedge connect c8y --test`
+When we validate stdout
+When we cleanup EnvironmentC8y
+Then we find a successful message in stdout
+Then the test has passed
+
+"""
+
+
+class RestartBridge(EnvironmentC8y):
```
Comment (albinsuresh) [inline tests/PySys/call_tedge_connect_c8y_--test_positive/run.py:20]: This class name doesn't really reflect what the test is doing.

File: tests/PySys/call_tedge_connect_c8y_--test_positive/run.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    def setup(self):
```
Comment (albinsuresh) [inline tests/PySys/call_tedge_connect_c8y_--test_positive/run.py:21]: Just FYI: If you don't really have any explicit "setup" logic, redeclaring the setup method here is optional. I see that you're registering a cleanup function in this setup. But, that cleanup is not doing anything either. So you could skip both these `setup` and `mycleanup` functions.

File: tests/PySys/call_tedge_connect_c8y_--test_negative/run.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+Given a configured system
+When we execute `sudo tedge connect c8y --test`
+When we validate stderr
+Then we find an error message in stderr
+Then test has passed
+"""
+
+
+class PySysTest(BaseTest):
```
Comment (albinsuresh) [inline tests/PySys/call_tedge_connect_c8y_--test_negative/run.py:14]: Although not mandatory, it's a common practice to name the test classes reflecting what it really does. It helps when you're trying to search for a specific test when we have a large number of tests. Here I'd have named the test as `TedgeConnectTestNegative` or something like that.

**Review by albinsuresh — CHANGES_REQUESTED** [review]
The test directory names like `call_tedge_connect_c8y_--test_negative` are really a mouthful. Short CamelCased names are the norm in the pysys community. So, I'd name those directories something like `TedgeConnectTestSuccess` and `TedgeConnectTestFailure` or something like that.

File: tests/PySys/call_tedge_connect_c8y_--test_negative/run.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from pysys.basetest import BaseTest
+
+"""
+Test `tedge connect c8y --test` (successful case):
```
Comment (abelikt) [inline tests/PySys/call_tedge_connect_c8y_--test_negative/run.py:4]: Better change that line to
"Run connection test without being connected (negative case):"

File: tests/PySys/call_tedge_connect_c8y_--test_positive/run.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import sys
+
+sys.path.append("environments")
+from environment_c8y import EnvironmentC8y
+
+"""
+Test `tedge connect c8y --test` (successful case):
```
Comment (abelikt) [inline tests/PySys/call_tedge_connect_c8y_--test_positive/run.py:7]: Maybe here also change that line to
"Run connection test while being connected (positive case):"
It's maybe a bit more understandable for people that don't know the precise command.

File: tests/PySys/call_tedge_connect_c8y_--test_positive/run.py (hunk 1 above)
Comment (rina23q) [reply tests/PySys/call_tedge_connect_c8y_--test_positive/run.py:20]: Will change it to `TedgeConnectTestPositive`.

Comment (abelikt) [reply tests/PySys/call_tedge_connect_c8y_--test_positive/run.py:20]: Hi @albinsuresh I would recommend to continue with snake case names as the other directory names are also snake case

File: tests/PySys/call_tedge_connect_c8y_--test_positive/run.py (hunk 2 above)
Comment (rina23q) [reply tests/PySys/call_tedge_connect_c8y_--test_positive/run.py]: good to know. Removed these two classes.

**Review by albinsuresh — APPROVED** [review]

**Review by abelikt — APPROVED** [review]
LVGTM

**Pull request merged**